sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobParallelUploadOptions.java

```diff
@@ -53,21 +53,32 @@ public BlobParallelUploadOptions(Flux<ByteBuffer> dataFlux) {
     /**
      * Constructs a new {@code BlobParalleUploadOptions}.
      *
-     * @param dataStream The data to write to the blob. The data must be markable. This is in order to support retries.
-     * If the data is not markable, consider opening a {@link com.azure.storage.blob.specialized.BlobOutputStream} and
-     * writing to the returned stream. Alternatively, consider wrapping your data source in a
-     * {@link java.io.BufferedInputStream} to add mark support.
-     * @param length The exact length of the data. It is important that this value match precisely the length of the
-     * data provided in the {@link InputStream}.
+     * Use {@link #BlobParallelUploadOptions(InputStream)} instead to supply an InputStream without knowing the exact
+     * length beforehand.
+     *
+     * @param dataStream The data to write to the blob.
+     * @param length The exact length of the data, or -1. It is important that a non-negative value match precisely the
+     * length of the data provided in the {@link InputStream}.
+     * @deprecated length is no longer necessary; use {@link #BlobParallelUploadOptions(InputStream)} instead.
      */
+    @Deprecated
     public BlobParallelUploadOptions(InputStream dataStream, long length) {
         StorageImplUtils.assertNotNull("dataStream", dataStream);
-        StorageImplUtils.assertInBounds("length", length, 0, Long.MAX_VALUE);
+        StorageImplUtils.assertInBounds("length", length, -1, Long.MAX_VALUE);
```

Comment:
should we add overload instead of using magic number and use nullable `Long` to represent this?

well, I just noticed overload below. In that case I'd keep original validation condition here and point users towards other ctor. or maybe create `(InputStream, Long)` ctor and move this logic there.

Reply:
Is the deprecation not enough to point users to the other ctor? I don't like the idea of introducing another ctor with a Long in it because the whole point of this change it's meaningless to provide one. There's nothing extra bought from providing a length, we're just preserving old behavior in case someone relied on us catching a mismatched length or something like that. A `Long` implies it may still be of use or provide some optimization or something when it really doesn't.

Reply:
then we should either have private ctor or don't call one from the other one. 
We should also deprecate getLength and replace it with something like `Long getSize()` ?

Reply:
I liked size in the context of expanding the ParallelTransferOptions back then because we could also rename the setter. But here we're not adding a new setter to complete the connection. Perhaps `Long getOptionalLength()`?

Reply:
either way is fine, we'll know at apireview.